waiter/src/waiter/handler.clj

```diff
@@ -1282,3 +1282,101 @@
+                              (-> (au/execute
+                                    (fn send-signal-to-instance []
+                                      (scheduler/signal-instance scheduler instance signal-type))
+                                    thread-pool)
+                                  async/<!
+                                  :result)]
+                          (if success
+                            (do
+                              (log/info "marking instance" instance-id "as killed")
+                              (counters/inc! (metrics/service-counter service-id "instance-counts" "killed"))
+                              ;; (counters/inc! (metrics/service-counter service-id "scaling" "scale-down" "success"))
+                              (scheduler/track-kill-candidate! instance-id :killed eject-backoff-base-time-ms)
+                              ;; (service/release-instance! populate-maintainer-chan! instance (result-map-fn :killed))
+                              (notify-instance-killed-fn instance))
+                            (do
+                              (log/info "failed kill attempt, releasing instance" instance-id)
+                              ;; (counters/inc! (metrics/service-counter service-id "scaling" "scale-down" "kill-fail"))
+                              (service/release-instance! populate-maintainer-chan! instance (result-map-fn :not-killed))))
+                          (when response-chan (async/>! response-chan kill-result))
+                          success))
+                    ;; (do
+                      ;; (log/info "no instance available to kill")
+                      ;; (counters/inc! (metrics/service-counter service-id "scaling" "scale-down" "unavailable"))
+                      ;; false)
+                               ))
+          (catch Exception ex
+            ;; (counters/inc! (metrics/service-counter service-id "scaling" "scale-down" "fail"))
+            (log/error ex "unable to send signal to instance " instance-id)
+            (when response-chan (async/>! response-chan {:instance-id instance-id :status http-500-internal-server-error :message (.getMessage ex)}))))))))
+
+
+
+(defn signal-handler
+  "Handler that supports sending signals to instances of a particular service on a specific router."
+  [notify-instance-killed-fn peers-acknowledged-eject-requests-fn allowed-to-manage-service?-fn scheduler populate-maintainer-chan! timeout-config
+   scale-service-thread-pool {:keys [route-params] {:keys [src-router-id]} :basic-authentication :as request}]
+  (let [{:keys [service-id]} route-params
+        {:strs [instance-id signal-type]} (-> request ru/query-params-request :query-params)
+        correlation-id (cid/get-correlation-id)]
+        
+    (log/info "received request to send" signal-type "to instance" instance-id "from" src-router-id)
+    (log/info "CORRELATION-ID: " correlation-id)
+    (async/go
```

Comment:
We should run authorization check using `allowed-to-manage-service?-fn` before trying to execute the signal.